### PR TITLE
memory: Fix DEBUG_HEAP build.

### PR DESCRIPTION
**Pull request by dabekjakub**
Rename function to fix DEBUG_HEAP build.

Signed-off-by: Jakub Dabek <jakub.dabek@intel.com>

### CODE REVIEW

File: src/include/sof/lib/alloc.h [hunk 1]
```diff
@@ -182,7 +182,7 @@ void rfree(void *ptr);
 	} while (0);					\
```
Comment (dbaluta) [inline src/include/sof/lib/alloc.h]: Please do not add a 'dot' at the end of the commit message. Also, please be more specific about the build failure. Maybe a small snippet of the build error.

Why renaming fixes the build?

Comment (dabekjakub) [reply src/include/sof/lib/alloc.h]: Dots removed. 
I do not know why we should go into why it is not building correctly - does it matter right now? 
Function under define was not named correctly so it had to be renamed.

Comment (dbaluta) [reply src/include/sof/lib/alloc.h]: @dabekjakub yes we really need to go into WHY we do changes. We need to develop good habits. Imagine if for such small changes we are not writing clear commit messages what are we going to do when we will have very complex design changes?!

So, please mention in the commit message why the build failed.

Comment (dabekjakub) [reply src/include/sof/lib/alloc.h]: @dbaluta what I mean mostly is that the change itself is self-explanatory.

The fact that the name of the function had to be changed in one place under a specific configuration clearly implies that it was defined differently in other parts of the code.

Comment (jajanusz) [reply src/include/sof/lib/alloc.h]: Yes description usually should tell 'why', for bigger/unclear stuff you need at least paragraph, but for simple change simple sentence is enough. We need to find good balance between being overly obsessive and not-discouraging developers

Comment (dbaluta) [reply src/include/sof/lib/alloc.h]: @dabekjakub 

> @dbaluta what I mean mostly is that the change itself is self-explanatory.
> 
> The fact that the name of the function had to be changed in one place under a specific configuration clearly implies that it was defined differently in other parts of the code.

@dabekjakub would like to see this in the commit message. It just a one line explanation.

I'm insisting on this to create a good habit on documenting the commit messages. I understand that it seems obvious for you why the change is need but it doesn't hurt to add the explanation in the commit message. 

For me it's not clear for example if the build fails with "multiple redefinition" of the same function or just actually function not defined.

**Pull request merged**